### PR TITLE
Label/tagging/deployment tweaks

### PR DESCRIPTION
**Pull request by Limess**
Few things here:

* Switch from label-schema.org labels to opencontainer labels, the former is defunkt (https://github.com/opencontainers/image-spec/blob/master/annotations.md)
* Add cache for go deps
* Remove dep install, it's pre-installed in the circle base image https://github.com/circleci/circleci-images/blob/b1075968c8279567cdb74625bf73f7963de4d17d/golang/generate-images
* Push image with the git SHA, do not build a new image if an image for that SHA already exists (i.e. the build has been re-ran)
* Add the date created and sha to the image labels. Also add the region and service name as runtime docker labels. These are all propagated to Splunk logs:
screen shot 2018-07-31 at 15 30 42
* Deploy to both regions after publishing. If eu-west-1 fails, us-east-1 will not be published. This is done via `aws cloudformation deploy` which first creates a changeset, then deploys it, waiting for the deploy to complete. The AWS credentials for this deploy are currently stored in a CircleCI context.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,18 +5,36 @@ defaults: &defaults
...
-          command: curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+      - restore_cache:
+          key: v1-gopkg-{{ checksum "Gopkg.lock" }}
 
       - run:
           name: Install dependencies
-          command: dep ensure -v
+          command: |
+            if [ ! -d  /go/src/github.com/Financial-Times/prometheus-health-check-exporter/vendor ]; then
+                dep ensure
```
Comment (lucas42) [inline .circleci/config.yml]: How does dep get installed?

File: Dockerfile [hunk 2]
```diff
@@ -28,4 +22,19 @@ COPY --from=build /tmp/health-check-exporter .
 
 EXPOSE 9842
```
Comment (lucas42) [inline Dockerfile]: Do we want to parameterise the port used somehow?

File: Makefile [hunk 3]
```diff
@@ -1,52 +1,102 @@
...
+	@if [[ -z "$(SPLUNK_HEC_TOKEN)" ]]; then echo "❌ $(RED)SPLUNK_HEC_TOKEN is not available. $(NO_COLOUR)This is a required variable for cloudformation deployments" && exit 1; fi
+
+deploy-stack: validate-aws-stack-command ## Create the cloudformation stack
+	@printf '%b\n' ">> $(TEAL)deploying cloudformation stack"
+	@aws cloudformation deploy \
+		--stack-name "mon-agg-ecs-service-$(REPO_NAME)" \
+		--template-file deployments/cloudformation.yml \
+		--parameter-overrides \
+			SplunkHecToken=$(SPLUNK_HEC_TOKEN) \
+			DockerRevision="$(DOCKER_TAG)" \
```
Comment (lucas42) [inline Makefile]: Doesn't this also require `ServiceName` as a parameter?  Or where does that come from?

File: deployments/cloudformation.yml [hunk 4]
```diff
@@ -71,11 +81,11 @@ Resources:
     Type: AWS::ElasticLoadBalancingV2::ListenerRule
     Properties:
       ListenerArn: !ImportValue "Prometheus-ECS-Cluster-HTTPS-Listener-Arn"
-      Priority: 5
+      Priority: 600
       Conditions:
         - Field: host-header
           Values:
-            - prometheus-health-check-exporter.in.ft.com
+            - !Sub "${ServiceName}.in.ft.com"
```
Comment (lucas42) [inline deployments/cloudformation.yml]: Do all our services' names match their hostnames?

File: deployments/cloudformation.yml [hunk 5]
```diff
@@ -71,11 +81,11 @@ Resources:
     Type: AWS::ElasticLoadBalancingV2::ListenerRule
     Properties:
       ListenerArn: !ImportValue "Prometheus-ECS-Cluster-HTTPS-Listener-Arn"
-      Priority: 5
+      Priority: 600
```
Comment (lucas42) [inline deployments/cloudformation.yml]: We need to have a think about what pattern we use for assigning priorities.  Was there a particular reason for changing this from 5 to 600?

File: .circleci/config.yml (hunk 1 above)
Comment (Limess) [reply .circleci/config.yml]: It's pre-installed on the circle base image (says in the PR description) https://github.com/circleci/circleci-images/blob/b1075968c8279567cdb74625bf73f7963de4d17d/golang/generate-images

File: Dockerfile (hunk 2 above)
Comment (Limess) [reply Dockerfile]: Methinks yes, or remove it altogether.

File: deployments/cloudformation.yml (hunk 4 above)
Comment (Limess) [reply deployments/cloudformation.yml]: probably not but I don't think this template can be properly generic as a service could have sidecards or similar.

File: deployments/cloudformation.yml (hunk 5 above)
Comment (Limess) [reply deployments/cloudformation.yml]: Raised https://github.com/Financial-Times/monitoring-aggregation-ecs/pull/2 to address this.

File: Makefile (hunk 3 above)
Comment (Limess) [reply Makefile]: It uses the default value for the param defined in the CF template at the moment.

**Pull request merged**